### ISSUE TITLE
Rework RFE operator

### ISSUE THREAD

**Issue by rhiever** [rhiever/tpot#280]
Currently, we use SVC in the Recursive Feature Elimination (RFE) operator to select features. RFE seems to be incredibly slow, and I rarely see it in the "best" pipelines on our benchmarks. Perhaps there is room for improvement.

We should consider using another estimator, such as the ExtraTreesClassifier (ETC), in place of SVC. Let's compare RFE w/ SVC to RFE with ETC and see how they compare.

We should also add `n_features_to_select` as an evolvable parameter. Not sure how that one got dropped.

Lastly, let's explore [RFECV](http://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.RFECV.html#sklearn.feature_selection.RFECV) as a possible replacement for RFE so CV can be used to choose `n_features_to_select` instead. I suspect that RFECV will be _very_ slow, but let's validate that assumption first.

cc @sahilshah1194

**Comment by weixuanfu2016** [comment]
RFE operator will cause CPU freezing (low CPU usage, like < 20%) when evaluating pipeline or fitting with a large dataset of a large number of features (like #271 ). Use the codes below can reproduce the CPU freezing.

```
import time
from sklearn import datasets
from sklearn.linear_model import LogisticRegression
from sklearn.cross_validation import train_test_split, cross_val_score
from sklearn.feature_selection import RFE
from sklearn.svm import SVC
from sklearn.naive_bayes import GaussianNB
from sklearn.pipeline import make_pipeline
X, y = datasets.make_classification(n_samples=10000, n_features=4000,
                                    n_informative=2, random_state=42)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.90,
                                                    random_state=42)
svc = SVC(kernel="linear", C=1)
rfe = RFE(estimator=svc, step=1)
gnb = GaussianNB()
sklearn_pipeline = make_pipeline(rfe,gnb)

time_start= time.time()
sklearn_pipeline.fit(X_train, y_train) # freezing here
cv_scores = cross_val_score( sklearn_pipeline, X_train, y_train, cv=3, scoring="accuracy")
time_end = time.time()
print('Simulation Dataset:',time_start-time_end)

```

**Comment by rhiever** [comment]
Could using SVC as the estimator be the issue? We should see if RFE will no longer freezes with ETC as the estimator.

**Comment by weixuanfu2016** [comment]
For RFE operator, argument `step` = 0.5 will speed up the process a lot, and no freezing happened with the simulated dataset with both SVC and ETC estimators. Also I tested step=0.1, 0.2, 0.3 and 0.4 with both estimators, and all tests passed without freezing. However I tested with the dataset from issue #271, CPU freezing happened with step = 0.5 when using SVC as estimators but no freezing happened when using ETC estimator. Maybe changing the estimator is the solution to avoid freezing.

**Comment by rhiever** [comment]
@weixuanfu2016, can you please verify that RFE w/ the ETC estimator and `step = 0.1` doesn't freeze with the data set from #271?

**Comment by weixuanfu2016** [comment]
Tested 10 times with ETC estimator and step = 0.1, no freezing happened and all tests finished in 10 seconds

**Comment by sahilshah1194** [comment]
I set up these code to see the differences between SVC and ETC across differents steps and it looks like ETC performs at least as good as SVC for steps from 0.1-0.7. 
```
import time
import numpy as np
from sklearn import datasets
from sklearn.linear_model import LogisticRegression
from sklearn.cross_validation import train_test_split, cross_val_score
from sklearn.feature_selection import RFE
from sklearn.svm import SVC
from sklearn.naive_bayes import GaussianNB
from sklearn.pipeline import make_pipeline
from sklearn.ensemble import ExtraTreesClassifier

X, y = datasets.make_classification(n_samples=10000, n_features=4000,
                                    n_informative=2, random_state=42)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.90,
                                                    random_state=42)

steps = np.arange(0.1,0.7,0.1)

for step in np.nditer(steps):
	print('-------------------------------------------------')
	print('Beginning setup for step {}'.format(step))
	svc = SVC(kernel="linear", C=1)
	rfe = RFE(estimator=svc, step=step)

	etc = ExtraTreesClassifier()
	rfe2 = RFE(estimator=etc, step=step)

	gnb = GaussianNB()
	gnb2 = GaussianNB()
	sklearn_pipeline = make_pipeline(rfe,gnb)
	sklearn_pipeline2 = make_pipeline(rfe2,gnb2)

	time_start= time.time()
	print('Fitting RFE with SVC')
	sklearn_pipeline.fit(X_train, y_train) # freezing here
	cv_scores = cross_val_score( sklearn_pipeline, X_train, y_train, cv=3, scoring="accuracy")

	print('Fitting RFE with ETC')
	sklearn_pipeline2.fit(X_train, y_train)
	cv_scores2 = cross_val_score( sklearn_pipeline2, X_train, y_train, cv=3, scoring="accuracy") 
	time_end = time.time()
	#print('Simulation Dataset:',time_end-time_start)
	print('Step: {}'.format(step))
	print('SCV: {}'.format(cv_scores))
	print('ETC: {}'.format(cv_scores2))
```

**Comment by rhiever** [comment]
@sahilshah1194, can you please plot that data and post the chart here?

**Comment by sahilshah1194** [comment]
![rfe_example](https://cloud.githubusercontent.com/assets/817331/19939314/c025036c-a0fe-11e6-874c-1aa95bcc6584.png)

Couldn't figure out how to add the labels in Excel for some reason: 
X-axis: Step value
Y-axis: Mean score
